### ISSUE TITLE
The difference between exec and query + possible bug

### ISSUE THREAD

**Issue by prk3** [blackbeam/mysql_async#129]
The documentation of `mysql_async` does not explain what is the differences between `exec` and `query` methods. I noticed that rows returned by `query_map` contain values represented as `Value::Bytes`, e.g.
```rust
let res = conn.query_map(
        String::from("SELECT a, b, c FROM table ORDER BY a"),
        |row: Row| row.unwrap().into_iter().collect::<Vec<Value>>()
    )
    .await?;
```
```
...
Bytes(
    "1",
),
...
``` 
whereas `exec_map` returns rows with more specific `Value` variants, e.g.
```rust
let res = conn.exec_map(
        "SELECT a, b, c FROM table ORDER BY a",
        (),
        |row: Row| row.unwrap().into_iter().collect::<Vec<Value>>()
    )
    .await?;
```
```
...
Int(
    1,
),
...
```
Is this intentional? What are the use cases of both methods?

**Comment by blackbeam** [comment]
Hi.

**tldr;** expected server behaviour.

`query*` methods are related to the so called "text protocol", and `exec*` methods, in their turn, to the so called "binary protocol". This distinction isn't mentioned in the mysql_async crate docs 
 (sorry for this), but is mentioned in the [mysql crate docs](https://docs.rs/mysql/19.0.1/mysql/#text-protocol).

The text protocol (see [here][tp]), from user's point of view, is just plain queries without parameters (or with insecure parameters, handcrafted on the client side). The result set of the text protocol is defined to be in [textual form][trs].

The binary protocol (see [here][bp]) relates to prepared statements and stored procedures execution. The result set of the binary protocol is defined to be in [binary form] [brs].

[tp]: https://dev.mysql.com/doc/internals/en/text-protocol.html
[trs]: https://dev.mysql.com/doc/internals/en/com-query-response.html#packet-ProtocolText::Resultset
[bp]: https://dev.mysql.com/doc/internals/en/prepared-statements.html
[brs]: https://dev.mysql.com/doc/internals/en/binary-protocol-resultset.html

**Comment by blackbeam** [comment]
> What are the use cases of both methods?

`exec*` - you must use it when your query have parameters (for safety reasons). You also may use it when your query have results (to avoid parsing from textual representation).
`query*` - when there is no parameters nor results in a query. Note, that `exec*` implies an implicit or explicit call to `prep` at least once, so it may be faster to parse a relatively small text result set than to invoke an `exec*` call.

**Comment by prk3** [comment]
Alright, that makes sense. Thanks for the explanation!
